llama_index/llm_predictor/base.py

```diff
@@ -132,22 +129,39 @@ def _log_template_data(
         ):
             pass
 
-    def predict(self, prompt: BasePromptTemplate, **prompt_args: Any) -> str:
+    def predict(self, prompt: BasePromptTemplate,
+        output_cls: Type[BaseModel] = None,
+                 **prompt_args: Any) -> str:
         """Predict."""
         self._log_template_data(prompt, **prompt_args)
-
         if self._llm.metadata.is_chat_model:
             messages = prompt.format_messages(llm=self._llm, **prompt_args)
             messages = self._extend_messages(messages)
-            chat_response = self._llm.chat(messages)
-            output = chat_response.message.content or ""
+            if output_cls is not None:
```

Comment:
I wonder if this can be done without modifying the LLM Predictor class

Technically, we've been meaning to remove this class for a while 😅